src/libtorch.cc

```diff
@@ -740,27 +756,34 @@ ModelInstanceState::ValidateInputs(const size_t expected_input_cnt)
       // input names since they are the keys for the dictionary
       input_index_map_[io_name] = i;
     } else {
-      // input tensor name must be in 'allowed_inputs' or must follow the naming
-      // convention
-      auto itr = allowed_inputs.find(io_name);
-      if (itr != allowed_inputs.end()) {
-        input_index_map_[io_name] = std::distance(allowed_inputs.begin(), itr);
-      } else {
-        try {
-          int start_pos = io_name.find(deliminator);
-          if (start_pos == -1) {
-            throw std::invalid_argument("input must follow naming convention");
+      switch (naming_convention) {
+        case FORWARD_ARGUEMENT: {
+          auto itr = allowed_inputs.find(io_name);
+          if (itr != allowed_inputs.end()) {
+            input_index_map_[io_name] =
+                std::distance(allowed_inputs.begin(), itr);
           }
-          ip_index = std::atoi(io_name.substr(start_pos + 2).c_str());
-          input_index_map_[io_name] = ip_index;
+          break;
         }
-        catch (std::exception& ex) {
-          return TRITONSERVER_ErrorNew(
-              TRITONSERVER_ERROR_INTERNAL,
-              ("input '" + io_name +
-               "' is neither an input argument to the model nor does it "
-               "follow the naming convention i.e. <name>__<index>.")
-                  .c_str());
+        case NAMED_INDEX: {
+          try {
+            int start_pos = io_name.find(deliminator);
+            if (start_pos == -1) {
+              throw std::invalid_argument(
+                  "input must follow naming convention");
+            }
+            ip_index = std::atoi(io_name.substr(start_pos + 2).c_str());
+            input_index_map_[io_name] = ip_index;
+          }
+          catch (std::exception& ex) {
+            // Already ran through this logic earlier. Try block will not catch
+            // an exception.
```

Comment:
Just remove the check as you know the I/Os are "good" after `GetNamingConvention()`